internal/repository/paymentservice/client.go

```diff
@@ -63,3 +65,20 @@ func (i Impl) AddTransaction(ctx context.Context, transaction Transaction) error
 	err := i.client.Perform(ctx, http.MethodPost, url, transaction, &response)
 	return errByStatus(err, response.Status)
 }
+
+func (i Impl) UpdateTransaction(ctx context.Context, transaction Transaction) error {
+	url := fmt.Sprintf("%s/v1/transactions", i.baseUrl)
+	response := aurestclientapi.ParsedResponse{}
+	err := i.client.Perform(ctx, http.MethodPut, url, transaction, &response)
+	return errByStatus(err, response.Status)
+}
+
+func (i Impl) GetTransactionByReferenceId(ctx context.Context, reference_id string) (Transaction, error) {
+	url := fmt.Sprintf("%s/v1/transactions?transaction_identifier=%s", i.baseUrl, url.QueryEscape(reference_id))
+	bodyDto := TransactionResponse{}
+	response := aurestclientapi.ParsedResponse{
+		Body: &bodyDto,
+	}
+	err := i.client.Perform(ctx, http.MethodGet, url, reference_id, &response)
+	return bodyDto.Payload[0], errByStatus(err, response.Status)
```

Comment:
bodyDto.Payload[0] will panic if Payload is nil or an empty slice.

So instead of the one line return statement, it gets a little more complicated:
```
err = errByStatus(err, response.Status)
if len(bodyDto.Payload) == 0 {
   if err == nil {
      err = NotFoundError
   }
   return Transaction{}, err
}
return bodyDto.Payload[0], err
```